app/services/teacher_training_api/import_subject.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module TeacherTrainingApi
+  class ImportSubject
+    include ServicePattern
+
+    def initialize(subject:)
+      @attrs = subject["attributes"]
+    end
+
+    def call
+      return unless code
```

Comment:
There is one subject 'Modern Languages' without a code - I've just skipped over that for now since surely it cannot be referenced in the API response for a course's `subject_codes`.

Incidentally, there is another subject 'Modern languages (other)' which does have a code.

Reply:
Could we store the id of the subject as well and use that in the `Subject.find_or_initialize_by`, something like `publish_id`, `teacher_training_id` or `remote_id`?

Reply:
We defs could do! But since it's the `subject_code` that is returned on the `course` from the API, I'm not sure what benefit that would bring?

Reply:
hmmm deferring to @gpeng deliberation 

If we do can we add the same for `subjects` and enforce it for register `course` import as well

https://github.com/DFE-Digital/teacher-training-api/blob/9f59717e39ec654db4a0f893281985a20fe5d5d2/db/migrate/20201113114331_add_uuid_ossp_extension.rb

Reply:
@gpeng what are your thoughts on this? ☝️

Reply:
🤔 Errrm... for subjects the codes should be reliable as they're pretty fixed and we control them. The courses have a UUID I think which might be useful as we have had providers ask for their course codes to be changed and this caused problems with Apply (and would probably do the same for us). The courses will also change over recruitment cycles, well, the code shouldn't usually but we haven't decided how we'll handle recruitment cycles yet so it would be good to have a way of uniquely identifying a course record.